plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductFormTemplates/BlockBasedTemplateInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductFormTemplates;
+
+/**
+ * Interface for block-based template.
+ */
+interface BlockBasedTemplateInterface extends BlockContainerInterface {
+	/**
+	 * Get a block by ID.
+	 *
+	 * @param string $block_id The block ID.
+	 */
+	public function get_block_by_id( string $block_id ): ?BlockInterface;
```

Comment:
We have talked about getting by block name as well.

Reply:
I know this is largely opinion, but I tend to prefer brevity where possible so my personal preference would be to stick with `get_block` for retrieving a singular block by ID as that also creates a similar API to the existing one in the client.

We also cannot create a `get_block_by_type` since we cannot limit that to response to a single block so this would be the only "get block" function.  We could create a separate method `get_blocks_by_type` if we need that in the future.